consensus/polybft/bridge_event_manager.go

```diff
@@ -542,24 +563,47 @@ func (b *bridgeEventManager) multicast(msg interface{}) {
 func (b *bridgeEventManager) GetLogFilters() map[types.Address][]types.Hash {
 	var bridgeMessageResult contractsapi.BridgeMessageResultEvent
 
+	var bridgeMsg contractsapi.BridgeMsgEvent
+
 	return map[types.Address][]types.Hash{
-		b.config.bridgeCfg.ExternalGatewayAddr: {types.Hash(bridgeMessageResult.Sig())},
+		b.config.bridgeCfg.ExternalGatewayAddr: {
+			types.Hash(bridgeMessageResult.Sig()),
+			types.Hash(bridgeMsg.Sig())},
 	}
 }
 
 // ProcessLog is the implementation of EventSubscriber interface,
 // used to handle a log defined in GetLogFilters, provided by event provider
 func (b *bridgeEventManager) ProcessLog(header *types.Header, log *ethgo.Log, dbTx *bolt.Tx) error {
-	bridgeMessageResultEvent := &contractsapi.BridgeMessageResultEvent{}
+	var (
+		bridgeMessageResultEvent contractsapi.BridgeMessageResultEvent
+		bridgeMsgEvent           contractsapi.BridgeMsgEvent
+	)
 
-	doesMatch, err := bridgeMessageResultEvent.ParseLog(log)
-	if err != nil {
-		return err
-	}
+	switch log.Topics[0] {
+	case bridgeMessageResultEvent.Sig():
+		doesMatch, err := bridgeMessageResultEvent.ParseLog(log)
+		if err != nil {
+			return err
+		}
 
-	if !doesMatch || b.sourceChainID != bridgeMessageResultEvent.SourceChainID.Uint64() {
-		return nil
-	}
+		if !doesMatch || b.externalChainID != bridgeMessageResultEvent.SourceChainID.Uint64() {
+			return nil
+		}
+
+		return b.state.BridgeMessageStore.removeBridgeEvents(&bridgeMessageResultEvent)
+	case bridgeMsgEvent.Sig():
+		doesMatch, err := bridgeMsgEvent.ParseLog(log)
+		if err != nil {
+			return err
+		}
 
-	return b.state.BridgeMessageStore.removeBridgeEventsAndProofs(bridgeMessageResultEvent)
+		if !doesMatch || b.internalChainID != bridgeMessageResultEvent.SourceChainID.Uint64() {
```

Comment:
Since this `ProcessLog` is the part of `EventSubscriber` in the `EventProvider`, which we use for internal (blade) originated events, we now this is an event from blade, but we need to check if this one is going to our external chain so that multiple bridge event managers don't insert the same internal originated event multiple times.

```suggestion
		if !doesMatch || b.externalChainID != bridgeMessageResultEvent.DestinationChainID.Uint64() {
```